framework/framework.go

```diff
@@ -142,3 +142,60 @@ func (f *Framework) ProcessDelete(obj interface{}, resources []Resource) error {
 
 	return nil
 }
+
+// ProcessUpdate is a drop-in for an informer's UpdateFunc. It receives the
+// custom object observed during TPR watches and anything that implements
+// Resource. ProcessUpdate takes care about all necessary reconciliation logic
+// for update events. For complex resources this means state has to be created,
+// deleted and updated eventually, in this order.
+//
+//     func updateFunc(oldObj, newObj interface{}) {
+//         err := f.ProcessUpdate(oldObj, newObj, resources)
+//         if err != nil {
+//             // error handling here
+//         }
+//     }
+//
+//     newResourceEventHandler := &cache.ResourceEventHandlerFuncs{
+//         UpdateFunc:    updateFunc,
+//     }
+//
+func (f *Framework) ProcessUpdate(oldObj, newObj interface{}, resources []Resource) error {
```

Comment:
I just started reviewing it while it's merged.

I don't think we should rely on on `oldObj`. This is specific to the cache util in the `client-go` lib.

I think in all our operators `OnUpdate` implementation should look like:

```
func OnUpdate(old, new interface{}) {
        OnAdd(new)
}
```

The operator should be able to reconcile only by retrieving state from the outside, and not rely on the fact that previous instance of TPO was actually reconciled. If that doesn't hold, then operator can not handle `OnAdd` after `ResyncPeriod` which is crucial IMO.